game-core/src/main/java/games/strategy/engine/framework/startup/ui/InGameLobbyWatcher.java

```diff
@@ -52,14 +53,16 @@ private InGameLobbyWatcher(
       final GameToLobbyConnection gameToLobbyConnection,
       final Consumer<String> errorReporter,
       final Consumer<String> reconnectionReporter,
-      @Nullable final InGameLobbyWatcher oldWatcher) {
+      @Nullable final InGameLobbyWatcher oldWatcher,
+      final Consumer<Boolean> connectivityHandler) {
```

Comment:
I think we might be a bit too heavy on callbacks here. At this point it might be better to have a single response object, notably 'errorReporter' and it handle the connectivity check failing. This is moot though if we do not combine postGame